src/test/java/org/broadinstitute/hellbender/engine/PathSpecifierUnitTest.java

```diff
@@ -0,0 +1,353 @@
+                // absolute reference to a file that contains a URI excluded character in the path ("#") which without
+                // encoding will be treated as a fragment delimiter
+                {FS_SEPARATOR + joinWithFSSeparator("project", "gvcf-pcr", "23232_1#1", "1.g.vcf.gz"),
+                        "file://" + getRootDirectoryAsURIPathString() + "project/gvcf-pcr/23232_1%231/1.g.vcf.gz", true, true},
+                // relative reference to a file on the local file system (relative to the current working directory)
+                {joinWithFSSeparator("path", "to", "localFile.bam"),
+                        "file://" + getCWDAsURIPathString() + "path/to/localFile.bam", true, true},
+                // Windows also accepts "/" as a valid root specifier
+                {"/", "file://" + getRootDirectoryAsURIPathString(), true, true},
+                {".", "file://" + getCWDAsURIPathString() + "./", true, true},
+                {"../.", "file://" + getCWDAsURIPathString() + ".././", true, true},
+                // an empty path is equivalent to accessing the current directory of the default file system
+                {"", "file://" + getCWDAsURIPathString(), true, true},
+
+                //***********************************************************
+                // Local file references using a URI with a "file://" scheme.
+                //***********************************************************
+
+                {"file:localFile.bam",              "file:localFile.bam",           true, false}, // absolute, opaque (not hierarchical)
+                {"file:/localFile.bam",             "file:/localFile.bam",          true, true},  // absolute, hierarchical
+                {"file://localFile.bam",            "file://localFile.bam",         true, false}, // file URLs can't have an authority ("localFile.bam")
+                {"file:///localFile.bam",           "file:///localFile.bam",        true, true},  // empty authority
+                {"file:path/to/localFile.bam",      "file:path/to/localFile.bam",   true, false},
+                {"file:/path/to/localFile.bam",     "file:/path/to/localFile.bam",  true, true},
+                // "path" looks like an authority, and will be accepted on Windows since it will be interpreted as a UNC authority
+                {"file://path/to/localFile.bam",    "file://path/to/localFile.bam", true, SystemUtils.IS_OS_WINDOWS},
+                // "localhost" is accepted as a special case authority for "file://" Paths on Windows; but not Linux
+                {"file://localhost/to/localFile.bam","file://localhost/to/localFile.bam", true, SystemUtils.IS_OS_WINDOWS},
+                {"file:///path/to/localFile.bam",   "file:///path/to/localFile.bam",    true, true},  // empty authority
+
+                //*****************************************************************************
+                // Valid URIs which are NOT valid NIO paths (no installed file system provider)
```

Comment:
Sigh. True, but its because only I have multiple branches with this code in different repos, and I'm trying to keep them as in-sync as possible in order to make propagating code review changes easier. In htjsdk-next, there are no providers for these so they're not valid there; when I moved them here I just changed the expected boolean to reflect that to make them pass. I'll have to reconcile them.